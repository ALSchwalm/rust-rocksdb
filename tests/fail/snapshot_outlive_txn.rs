use rocksdb::TxnDB;

fn main() {
    let db = TxnDB::open_default("foo").unwrap();
    let _snapshot = {
        let txn = db.txn();
        txn.snapshot()
    };
}