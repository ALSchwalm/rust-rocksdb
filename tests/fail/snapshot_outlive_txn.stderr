error[E0597]: `txn` does not live long enough
 --> tests/fail/snapshot_outlive_txn.rs:7:9
  |
5 |     let _snapshot = {
  |         --------- borrow later stored here
6 |         let txn = db.txn();
7 |         txn.snapshot()
  |         ^^^ borrowed value does not live long enough
8 |     };
  |     - `txn` dropped here while still borrowed
